### PR TITLE
:es6: getCreativeIds

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Because `dfpEnv` is central to everything in our commercial code, I think it's a good idea to define it properly.

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js [hunk 1]
```diff
@@ -2,7 +2,24 @@
 import { getUrlVars } from 'lib/url';
 import config from 'lib/config';
 
-const dfpEnv = {
+type Map<A, B> = { [A]: B };
+
+type DfpEnv = {
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js]: is it worth exporting the `type` too?

**Review by rich-nguyen — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js (hunk 1 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js]: 👍  Yes, absolutely

**Comment by regiskuckaertz** [comment]
@sndrs arf, tests fail on type declaration. Any idea how to solve this?

**Comment by sndrs** [comment]
i have no idea at all i'm afraid, i've never encountered exporting types!

**Comment by sndrs** [comment]
so that's tranpsilation failing, not the tests. which i imagine means strip flow types plugin cannot understand the syntax?

**Comment by SiAdcock** [comment]
Have you tried defining `advertIds` as an [indexer property](https://flow.org/en/docs/types/objects/#toc-objects-as-maps)

```js
advertIds: { [indexer: string]: number }
```

See also the [`flow-strip-types` test fixtures](https://github.com/babel/babel/blob/8434f89bc040897e1fa8f098dad0f5e236961ca8/packages/babel-plugin-transform-flow-strip-types/test/fixtures/strip-types/strip-type-annotations/actual.js#L40) on the babel repo

**Comment by regiskuckaertz** [comment]
I see our own @janua has already [reported this](https://github.com/babel/babel/issues/5697) to the babel team

**Comment by SiAdcock** [comment]
Nice! Naming the indexer seems to work, if you need a quick fix.

**Comment by regiskuckaertz** [comment]
@SiAdcock thanks, mate 😗

File: static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js [hunk 2]
```diff
@@ -2,6 +2,8 @@
 import { getUrlVars } from 'lib/url';
 import config from 'lib/config';
 
+type Map<K, V> = { [indexer: K]: V };
```
Comment (rich-nguyen) [inline static/src/javascripts/projects/commercial/modules/dfp/dfp-env.js]: nice use of pinky death grip

**Pull request merged**